### PR TITLE
Enhanced mine command

### LINKED ISSUE #816: I cant use `mine diamonds` literally unplayable
## Describe your suggestion
please i need this. Also some more general mining keywords like coal, iron, gold etc.

## Settings
Add `mine diamonds` / `mine diamond` as an alias of `mine diamond_ore`, `mine iron` as `mine iron_ore` etc.

## Context
You would become cooler than your toenails.

## Final checklist
- [x] I have not used any OwO's or UwU's in this issue.

### PR DESCRIPTION
**Pull request by Graicc**
* Added an alias map, that allows you to use shorthand for certain blocks (e.x. "diamond" -> "diamond_ore"). Currently just has ores.
* Blocks / aliases can be typed with a tailing s, which will be ignored (e.x "logs" -> "log"). This does not affect things that should have a tailing s like glass, nor aliases with tailing s's.

Resolves #816

### CODE REVIEW

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 1]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
                 logDirect("Will mine " + quantity + " " + blockTypes[0]);
                 return true;
             } catch (NumberFormatException | ArrayIndexOutOfBoundsException | NullPointerException ex) {}
-            for (String s : blockTypes) {
+
+            Map<String, String> aliases = new HashMap(); // This should definitely be in a setting i'm just too dumb for that
+            aliases.put("diamond", "diamond_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 2]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("iron", "iron_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 3]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("coal", "coal_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 4]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("lapis", "lapis_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 5]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("gold", "gold_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 6]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("redstone", "redstone_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 7]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("emerald", "emerald_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 8]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+            aliases.put("quartz", "quartz_ore");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bruh

**Comment by babbaj** [comment]
🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍 🦍

**Comment by Graicc** [comment]
Thank you for the insightful feedback

**Comment by StijnSimons** [comment]
Holy shit i just got aids

**Comment by 0-x-2-2** [comment]
hardcoded names. bruh

**Comment by metroid112** [comment]
I don't have PHD in programming but couldn't just accept names if "name + _ore" is a block?

**Comment by StijnSimons** [comment]
The thing you're missing is: just use diamond_ore???????

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 9]
```diff
@@ -511,7 +511,32 @@ public boolean isInGoal(int x, int y, int z) {
...
+
+            for (int i = 0; i < blockTypes.length; i++) {
+                String s = blockTypes[i];
+
+                s = BlockUtils.stringToBlockNullable(s) == null // Is it not already a block?
```
Comment (leijurv) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: this is 

insane

**Comment by LoganDark** [comment]
dude

**Comment by LoganDark** [comment]
just use an if statement
you don't need to use 5 nested ternaries

**Comment by LoganDark** [comment]
This will all be obsoleted once my commands branch gets merged, message me on Discord for more info @Cuteness#4357

**Comment by leijurv** [comment]
crazy

**Pull request closed without merging**